storage-proto/src/convert.rs

```diff
@@ -728,11 +737,28 @@ impl TryFrom<tx_by_addr::TransactionError> for TransactionError {
     fn try_from(transaction_error: tx_by_addr::TransactionError) -> Result<Self, Self::Error> {
         if transaction_error.transaction_error == 8 {
             if let Some(instruction_error) = transaction_error.instruction_error {
+                // Decode the index.
+                //
+                // Index values are 32-bit integers of the form:
+                // TTTTTTTT IIIIIIII xxxxxxxx xxxxxxxx
+                //
+                // * T - The top level index of the instruction that errored
+                // * I - The inner index of the instruction that errored; 0 if None, 1-indexed otherwise
+                let [outer_instruction_index, maybe_inner_instruction_index, _unused1, _unused2] =
+                    instruction_error.index.to_le_bytes();
+                let inner_instruction_index = maybe_inner_instruction_index.checked_sub(1);
```

Comment:
I've implemented this as a single-byte option, where 0 means `None` and 1 means 0. The implication is that this can only store an inner instruction index of at most 255. If we can CPI that many instructions, it probably means that Solana has undergone major changes, and all of this code is already gone.